test/OpenIddict.Server.Tests/OpenIddictServerBuilderTests.cs

```diff
@@ -0,0 +1,2000 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+using Moq;
+using OpenIddict.Abstractions;
+using Xunit;
+using static OpenIddict.Server.OpenIddictServerEvents;
+
+namespace OpenIddict.Server.Tests
+{
+    public class OpenIddictServerBuilderTests
+    {
+        [Fact]
+        public void Constructor_ThrowsAnExceptionForNullServices()
+        {
+            // Arrange
+            var services = (IServiceCollection)null;
+
+            // Act and assert
+            var exception = Assert.Throws<ArgumentNullException>(() => new OpenIddictServerBuilder(services));
+
+            Assert.Equal("services", exception.ParamName);
+        }
+
+        private class CustomContext : BaseContext
+        {
+            /// <summary>
```

Comment:
You can remove the XML comments, they are not very useful in this case.

Reply:
Resharper generated those when I used its 'generate missing members' functionality. I can remove them if you like, but it didn't take any time to add them either :).